pkg/pool-weighted/test/ManagedPool.test.ts

```diff
@@ -68,18 +67,18 @@ describe('ManagedPool', function () {
     await allTokens.approve({ from: owner, to: vault });
   });
 
-  async function deployPool(overrides: RawWeightedPoolDeployment = {}): Promise<WeightedPool> {
+  async function deployPool(overrides: RawManagedPoolDeployment = {}): Promise<ManagedPool> {
     const params = {
       vault,
       tokens: poolTokens,
       weights: poolWeights,
       owner: owner.address,
       aumFeeId: ProtocolFee.AUM,
-      poolType: WeightedPoolType.MOCK_MANAGED_POOL,
+      mockContractName: 'MockManagedPool',
```

Comment:
We moved away from the contract name in favor of an enum in https://github.com/balancer-labs/balancer-v2-monorepo/pull/2078.

This is not a very strong opinion, but I think the enum is a bit more accurate to select mock contracts (the comment applies here and wherever `mockContractName` is used). 

Given that the pool type is the object that we are creating, perhaps `poolType` is no longer the best name to select which contract to create.

Reply:
The downstream code is directly using that name in the `deploy` call, and it's not always the same. There's another mock for the settings, which gets passed elsewhere. This let me get rid of the enum completely. I thought it was ok to pass the string in this case, since it was semantic and directly used: literally the name of the mock contract to deploy.

Reply:
I'd agree that changing this to a string enum would be helpful.